### PR TITLE
tighten yara signature patterns

### PR DESCRIPTION
**Pull request by deadbits**
# Pull Request

## Description

Tightens YARA signature patterns to avoid potential false positives.

Analysis was performed by running the signatures over the entire OpenClawd `skills` repository and tuning to remove hits on benign text.

**Results from Openclawd skills repo:**

Some of these are still almost certainly false positives, but it's significantly lower than it was before. In some cases, single strings like `perfect` were causing signatures to trigger.

After tuning:
```bash
autonomy_abuse_generic                         files:     1  strings:     4
capability_inflation_generic                   files:     4  strings:     5
clawhavoc_campaign                             files:   381  strings:  5700
code_execution_generic                         files:    36  strings:   168
coercive_injection_generic                     files:    23  strings:    29
command_injection_generic                      files:    29  strings:    70
credential_harvesting_generic                  files:    28  strings:    32
indirect_prompt_injection_generic              files:     5  strings:     6
prompt_injection_generic                            files:     5  strings:     5
prompt_injection_unicode_steganography         files:     9  strings:    23
script_injection_generic                       files:     6  strings:    11
sql_injection_generic                          files:    15  strings:    24
system_manipulation_generic                    files:    25  strings:    30
tool_chaining_abuse_generic                    files:    28  strings:    29
```

Before tuning:
```
capability_inflation_generic            files:  833  strings:  1469
system_manipulation_generic             files:  311  strings:  609
prompt_injection_generic                files:  270  strings:  495
indirect_prompt_injection_generic       files:  252  strings:  577
credential_harvesting_generic           files:  244  strings:  507
code_execution_generic                  files:  194  strings:  1425
command_injection_generic               files:  177  strings:  530
tool_chaining_abuse_generic             files:  111  strings:  172
coercive_injection_generic              files:  43   strings:  54
script_injection_generic                files:  27   strings:  33
sql_injection_generic                   files:  15   strings:  24
prompt_injection_unicode_steganography  files:  9    strings:  23
autonomy_abuse_generic                  files:  3    strings:  6
```

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Test coverage improvement

## Related Issues

Closes #[issue number]
Fixes #[issue number]
Related to #[issue number]

## Changes Made

- Change 1: [describe]
- Change 2: [describe]
- Change 3: [describe]

### Test Coverage

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] All tests pass locally
- [ ] Test coverage maintained or improved

### Manual Testing

Describe manual testing performed:

```bash
# Commands run for testing
skill-scanner scan examples/test-skill
```

**Results:**
- Expected: [describe]
- Actual: [describe]

### Code Quality
- [ ] Code follows project style guidelines
- [ ] Type hints added where applicable
- [ ] Docstrings added/updated for public APIs
- [ ] No hardcoded credentials or secrets
- [ ] Error handling is comprehensive
- [ ] Logging is appropriate

### Documentation
- [ ] README updated (if needed)
- [ ] API documentation updated (if needed)
- [ ] CHANGELOG updated
- [ ] Code comments added for complex logic

### Security
- [ ] No new security vulnerabilities introduced
- [ ] Input validation added where needed
- [ ] Follows security best practices from workspace rules
- [ ] No eval/exec on user input without sanitization

### Testing
- [ ] Tests pass: `uv run pre-commit run --all-files`
- [ ] Benchmark passes: `uv run python evals/benchmark_runner.py`
- [ ] No regressions in existing functionality
- [ ] Edge cases covered

## Performance Impact

- [ ] No significant performance regression
- [ ] Performance benchmarks run (if applicable)
- [ ] Resource usage is acceptable

## Screenshots (if applicable)

Add screenshots or output examples if relevant.

## Additional Notes

Any additional information reviewers should know.

## Reviewer Checklist

For reviewers:
- [ ] Code changes are clear and well-documented
- [ ] Tests are comprehensive
- [ ] No security issues introduced
- [ ] Performance is acceptable
- [ ] Documentation is updated

### CODE REVIEW

File: skill_scanner/data/yara_rules/tool_chaining_abuse_generic.yara [hunk 1]
```diff
@@ -30,29 +30,38 @@ rule tool_chaining_abuse_generic{
...
-        // Read secret env var then send to network
-        $env_var_exfil = /\b(os\.environ|getenv|process\.env)[^.]{0,30}(SECRET|PRIVATE|KEY|TOKEN|PASSWORD|CREDENTIAL)[^.]{0,100}\b(requests\.(post|get)|urllib|fetch|curl|wget)\b/i
+        // Read secret env var then send to network - TIGHTENED
+        // Only match genuinely suspicious patterns: SECRET, PASSWORD, PRIVATE_KEY
+        // Not generic KEY/TOKEN which appears in normal API auth code
+        $env_var_exfil = /\b(os\.environ|getenv|process\.env)[^.]{0,30}(SECRET|PRIVATE|PASSWORD|CREDENTIAL)[^.]{0,100}\b(requests\.(post|get)|urllib|fetch|curl|wget)\b/i
 
         // === Exclusions ===
-        $security_docs = /\b(MITRE|ATT&CK|threat\s+(model|hunt)|detection\s+rule)/i
+        $security_docs = /\b(MITRE|ATT&CK|threat\s+(model|hunt)|detection\s+rule|security[_\s]?(scan|check|audit|monitor|guard|pattern|review)|threat[_\s]?pattern|vulnerability[_\s]?pattern|attack[_\s]?(vector|surface|pattern|example)|exfiltration[_\s]?(detect|prevent|pattern)|data[_\s]?loss[_\s]?prevent|YARA|prompt[_\s]?injection)\b/i
```
Comment (vineethsai7) [inline skill_scanner/data/yara_rules/tool_chaining_abuse_generic.yara]: This is a good one

**Review by vineethsai7 — APPROVED** [review]
